src/event/sys/unix.rs

```diff
@@ -11,6 +12,45 @@ use crate::{
     ErrorKind, Result,
 };
 
+pub struct CancelTx(FileDesc);
+
+impl CancelTx {
+    /// Release the cancellation token.
+    pub fn release(self) -> Result<()> {
+        // DO NOT write more than 1 byte. See try_read & WAKE_TOKEN
+        // handling - it reads just 1 byte. If you write more than
+        // 1 byte, lets say N, then the try_read will be woken up
+        // N times.
+        self.0.write(&[0x57])?;
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct CancelRx(pub(crate) Arc<FileDesc>);
+
+/// Construct a cancellation pair.
+pub fn cancellation() -> Result<(CancelTx, CancelRx)> {
```

Comment:
Is this supposed to call this from outside the scope, maybe pub(crate)?

Reply:
Not exposing this raises a warning that `CancelRx` is never constructed since it's present in another pub API, albeit in a `Option<&CancelRx>`, so it's typically used with `None`.

Reply:
I bumped the branch, feel free to try it out yourself and see if you can find a better solution!